CIP-transaction-inputs-as-list/README.md

```diff
@@ -0,0 +1,128 @@
+    TX --> A1((Output 1))
+    TX --> A2((Output 2))
+    TX --> A3((Output 3))
+```
+However, in this approach, the minting or staking validator requires a list of indices to be passed from the redeemer to select the correct inputs within the transaction input list. One vulnerability of this implementation is that the list of indices must be unique to prevent double validation, which could allow someone to unlock inputs without proper validation.
+
+To mitigate such attacks, one approach is to compute the uniqueness of this index list, but this can result in costly computations if the business logic requires validation of a large number of inputs. A simpler approach to mitigate these hacks is to offer flexibility by explicitly allowing the order of inputs to be determined by a transaction builder instead of the ledger. This eliminates the need to pass indexes through the redeemer.
+
+## Specification
+As per protocol specifications, the transaction body is structured as follows:
+```
+transaction_body =
+ { 0 : set<transaction_input>    ; inputs
+ , 1 : [* transaction_output]
+ , 2 : coin                      ; fee
+ , ? 3 : uint                    ; time to live
+ , ? 4 : [* certificate]
+ , ? 5 : withdrawals
+ , ? 6 : update
+ , ? 7 : auxiliary_data_hash
+ , ? 8 : uint                    ; validity interval start
+ , ? 9 : mint
+ , ? 11 : script_data_hash       ; New
+ , ? 13 : set<transaction_input> ; Collateral ; new
+ , ? 14 : required_signers       ; New
+ , ? 15 : network_id             ; New
+ }
+```
+
+Specifically, the inputs are currently represented as a set:
+```
+0 : set<transaction_input>    ; inputs
+```
+
+The proposed solution suggests modifying the inputs to a list format:
+```
+0 : [* transaction_input]    ; inputs
+```
+
+
+
+## Rationale: how does this CIP achieve its goals?
+The motivation behind this CIP stems from the observed limitations and inefficiencies associated with the current lexicographical ordering of transaction inputs. Currently, the strict lexicographical ordering necessitates the passing of index numbers through the redeemer to select the appropriate input, leading to potential inefficiencies and vulnerabilities to vector attacks.
+
+To mitigate these issues, the proposed solution suggests transitioning from a set-based representation of transaction inputs to a list-based representation.
+
+This CIP tries to revive the original draft [CIP-0051](https://github.com/cardano-foundation/CIPs/pull/231)
+
+### Alternatives
+#### 1. Retain the existing set-based representation with additional validation mechanisms: 
+
+This approach involves maintaining the current set-based representation of transaction inputs while implementing additional validation mechanisms to guarantee the uniqueness of index lists.
+However, it was determined that such approaches could introduce complexity and computational overhead
+
+#### 2. Hash the spending outref and use it for datum tagging: 
```

Comment:
also unsure what this means. examples would help

Reply:
The other workaround is tagging the output with the hash of the outref, I can update the CIP to clarify with a PlutusTX example, let me know.

https://plutus.readthedocs.io/en/latest/reference/writing-scripts/common-weaknesses/double-satisfaction.html?highlight=double#unique-outputs

Reply:
Yes @solidsnakedev I think example(s) will help here (and generally everywhere). 🤓

Reply:
Resolving, since original unclear statements is gone & further clarification provided in general.